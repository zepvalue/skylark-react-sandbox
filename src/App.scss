.App {
  text-align: center;
  display:flex;
  flex-direction: column;
}

.App-logo {
  height: 40vmin;
}

.App-header {
  background-image: url('./assets/spadeckchairs.jpg');
  background-repeat: no-repeat;
  background-size: 100% 100%;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

#App-Container {
  width: 100%;
  height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.App-link {
  color: #09d3ac;
}

.card {
  box-shadow: 5px 10px #888888;
  border-radius: 5px;
  border: 1px solid black;
  margin-bottom: 2rem;
}

#header_container {
  display:flex;
  flex-direction: column;
  
}

#header_text {
  font-size: 3rem;
  text-shadow: 0 0 5px rgba(0, 0, 0, 0.5);
}

#header-description {
  font-size: 1.0rem;
  text-shadow: 0 0 5px rgba(0, 0, 0, 0.5);
}

#form_container {
  border: 1px solid rgba(255, 255, 255, 0.35);
  background: rgba(255, 255, 255, 0.25);
  border-radius: 3px;
  height: 40px;
  padding:10px;
  display: flex;
  margin: auto;
  flex-direction: row;
  justify-content: center;
}

#search-button {
  text-transform: none;
  color: #fff;
  text-align: center;
  background: #e66756;
}

.DateRangePickerInput_arrow {
  background-color: #fff;
  margin-bottom: 20px;
}

.DateInput {
  height: 30px;
  display: inline-flex !important;
}

.DateInput_input {
  height:5px;
  cursor: pointer;
}

.DateRangePickerInput {
  background-color: #fff;
  display: inline-block;
  height: 30px;
}
