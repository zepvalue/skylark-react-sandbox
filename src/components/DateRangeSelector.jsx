import React, { useState } from 'react';
import { DateRangePicker } from 'react-dates';
import moment from 'moment';

const DateRangeSelector = ({ setDate }) => {
  const [startDate, setStartDate] = useState();
  const [endDate, setEndDate] = useState();
  const [focusedInput, setfocusedInput] = useState();

  function handleDataChange({ startDate, endDate }) {
    const formatStartDate = moment(startDate).format('YYYY-MM-DD');
    const formatEndDate = moment(endDate).format('YYYY-MM-DD');
    setDate({ formatStartDate, formatEndDate });
    setStartDate(startDate);
    setEndDate(endDate);
  }

  function handleFocusChange(focusedInput) {
    setfocusedInput(focusedInput);
  }

  function isOutsideRange() {
    return false;
  }
  return (
    <DateRangePicker
      endDate={endDate}
      endDateId="endDate"
      focusedInput={focusedInput}
      isOutsideRange={() => isOutsideRange()}
      onDatesChange={e => handleDataChange(e)}
      onFocusChange={e => handleFocusChange(e)}
      startDate={startDate}
      startDateId="startDate"
    />
  );
};

export default DateRangeSelector;
