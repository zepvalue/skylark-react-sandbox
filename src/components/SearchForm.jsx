import React, { useState } from 'react';
import axios from 'axios';
import DateRangeSelector from './DateRangeSelector.jsx';
import 'react-dates/initialize';

import 'react-dates/lib/css/_datepicker.css';

const SearchForm = ({ list, setResponseData }) => {
  const [hotelCode, setHotel] = useState('');
  const [date, setDate] = useState('');
  const [passengers, setPassengers] = useState(1);
  const [rooms, setRooms] = useState(1);

  const options = list.map((el, i) => (
    <option key={`key${i}`} value={el.code}>
      {el.name}
    </option>
  ));

  const optionsRooms = [1, 2, 3, 4].map((el, i) => {
    if (el === 1)
      return (
        <option key={`Room-${i}`} value={el}>
          {`${el} Room`}
        </option>
      );
    if (el > 1)
      return (
        <option key={`Rooms-${i}`} value={el}>
          {`${el} Rooms`}
        </option>
      );
  });

  const optionsPassengers = [1, 2, 3, 4, 5, 6, 7, 8].map((el, i) => {
    if (el === 1)
      return (
        <option key={`Person-${i}`} value={el}>
          {`${el} Person`}
        </option>
      );
    if (el > 1)
      return (
        <option key={`People-${i}`} value={el}>
          {`${el} People`}
        </option>
      );
  });

  function fetchAvailability() {
    const { formatStartDate, formatEndDate } = date;
    const FULL_URL = `http://gateway.skylark.com/ex/merged_hotel_stays/${hotelCode}?check_in=${formatStartDate}&check_out=${formatEndDate}&roomCount=${rooms}&passengerCount=${passengers}`;

    console.log(FULL_URL);
    const response = axios.get(FULL_URL);
    response.then(res => {
      if (res.data.rooms.length === 0) setResponseData('Sorry no availability');
      else setResponseData(`$ ${res.data.rooms[0].rates[0].pricing.totalPrice}`);
    });
  }

  return (
    <div id="form_container">
      <DateRangeSelector setDate={setDate} />
      <select value={hotelCode} onChange={e => setHotel(e.target.value)}>
        {options}
      </select>
      <select id="passengers" value={passengers} onChange={e => setPassengers(e.target.value)}>
        {optionsPassengers}
      </select>
      <select id="rooms" value={rooms} onChange={e => setRooms(e.target.value)}>
        {optionsRooms}
      </select>

      <button id="search-button" type="button" onClick={() => fetchAvailability()}>
        Search
      </button>
    </div>
  );
};

export default SearchForm;
