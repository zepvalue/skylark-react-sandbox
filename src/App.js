import React, { useState } from 'react';
import './App.scss';

import destinationData from 'data/destination-cancun.json';
import SearchForm from 'components/SearchForm';

const App = () => {
  const [responseData, setResponseData] = useState(null);
  const { hotels } = destinationData;
  return (
    <div className="App">
      <header className="App-header">
        <div id="App-Container">
          <div id="header_container">
            <div id="header_text">Booking is just the beginning</div>
            <div id="header_description">Skylark combines real time technology with human expertise.</div>
          </div>
          <SearchForm list={hotels} setResponseData={setResponseData} />
          <div>{responseData}</div>
        </div>
      </header>
    </div>
  );
};

export default App;
