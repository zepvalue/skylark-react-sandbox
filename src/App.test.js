import React, { useState } from 'react';
import { mount, configure, shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import renderer, { create } from 'react-test-renderer';
import mockAxios from 'axios';
import { act } from 'react-dom/test-utils';
import App from './App';
import DateRangeSelector from './components/DateRangeSelector.jsx';

import SearchForm from './components/SearchForm';

configure({ adapter: new Adapter() });

describe('React Unit Tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Button Snapshot', () => {
    const component = renderer.create(<button />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('deep rendering without crashing', () => {
    const props = {
      list: [{ name: 'room1' }, { name: 'room2' }, { name: 'room3' }],
      setResponseData() {
        return 'data';
      },
    };
    const wrapper = shallow(<SearchForm {...props} />);
    expect(wrapper.find(DateRangeSelector).length).toEqual(1);
    expect(wrapper.find(<select />).length).toEqual(0);
  });
});
