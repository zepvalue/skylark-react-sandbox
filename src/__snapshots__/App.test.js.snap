// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Unit Tests Button Snapshot 1`] = `<button />`;
